sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py

```diff
@@ -0,0 +1,124 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import (  # pylint: disable=unused-import
+    Union,
+    Any,
+    IO,
+    TYPE_CHECKING,
+)
+import six
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._policies import CognitiveServicesCredentialPolicy
+from ._response_handlers import (
+    prepare_receipt_result,
+    prepare_layout_result,
+
+)
+from ._generated.models import AnalyzeOperationResult
+from ._helpers import get_content_type, POLLING_INTERVAL
+if TYPE_CHECKING:
+    from ._credential import FormRecognizerApiKeyCredential
+
+
+class FormRecognizerClient(object):
+    """FormRecognizerClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of FormRecognizerApiKeyCredential if using an API key.
+    :type credential: ~azure.ai.formrecognizer.FormRecognizerApiKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, FormRecognizerApiKeyCredential, Any) -> None
+        self._client = FormRecognizer(
+            endpoint=endpoint,
+            credential=credential,
+            authentication_policy=CognitiveServicesCredentialPolicy(credential),  # TODO: replace with core policy
+            **kwargs
+        )
+
+    @distributed_trace
+    def begin_extract_receipts(self, form, **kwargs):
+        # type: (Union[str, IO[bytes]], Any) -> LROPoller
+        """Extract field text and semantic values from a given receipt document.
+        The input document must be of one of the supported content types - 'application/pdf',
+        'image/jpeg', 'image/png' or 'image/tiff'. Alternatively, use 'application/json'
+        type to specify the location (Uri) of the document to be analyzed.
+
+        :param form: .json, .pdf, .jpg, .png or .tiff type file stream.
+        :type form: str or stream
+        :keyword bool include_text_details: Include text lines and element references in the result.
+        :keyword str content_type: Media type of the body sent to the API.
+        :return: LROPoller
+        :rtype: ~azure.core.polling.LROPoller
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        include_text_details = kwargs.pop("include_text_details", False)
+        content_type = kwargs.pop("content_type", None)
+
+        if isinstance(form, six.string_types):
```

Comment:
As discussed in the UX study feedback, we probably want to split the "analyze from URL" and "analyze from client data" functions (the semantics are different enough that it probably is worth it). Not blocking this merge, though.

Reply:
Yes. I believe that @KrzysztofCwalina was convincingly enough when explaining why this wouldn't be an overload in .NET.

Reply:
Brought this up in our internal FR chat so that we can align across languages.